pkgs/development/libraries/libqmi/default.nix

```diff
@@ -1,23 +1,36 @@
-{ stdenv, fetchurl, pkgconfig, glib, python, libgudev, libmbim }:
+{ stdenv, fetchurl, pkgconfig, glib, python3, libgudev, libmbim }:
 
 stdenv.mkDerivation rec {
-  name = "libqmi-1.20.2";
+  pname = "libqmi";
+  version = "1.22.2";
 
   src = fetchurl {
-    url = "https://www.freedesktop.org/software/libqmi/${name}.tar.xz";
-    sha256 = "0i6aw8jyxv84d5x8lj2g9lb8xxf1dyad8n3q0kw164pyig55jd67";
+    url = "https://www.freedesktop.org/software/libqmi/${pname}-${version}.tar.xz";
+    sha256 = "09w20dsgr16bgbqw5ds7r6j2s6ihwyalh9zpbjhcn7cvm0afbwgi";
   };
 
   outputs = [ "out" "dev" "devdoc" ];
 
-  preBuild = ''
+  preConfigure = ''
     patchShebangs .
```

Comment:
This should be more specific to avoid overpatching (not sure if this is actually a problem) and make reasoning about this cleaner. Also, this is usually done in `postPatch`.

Reply:
Only files I see that might need this are
* https://gitlab.freedesktop.org/mobile-broadband/libmbim/blob/master/build-aux/mbim-codegen/mbim-codegen
* https://gitlab.freedesktop.org/mobile-broadband/libqmi/blob/master/build-aux/qmi-codegen/qmi-codegen
* https://gitlab.freedesktop.org/mobile-broadband/libqmi/blob/master/utils/qmidb/qmidb.py

Tried dropping it entirely and didn't have an issue building.